default.nix

```diff
@@ -90,6 +95,8 @@ with stdenv.lib; let
 
   commonBuildInputs = [ _nodejs makeWrapper ];  # TODO: git?
 
+  normalizeName = builtins.replaceStrings [ "@" "." "/" ] [ "_" "_" "_" ];
```

Comment:
Would be nice to have some documentation (or comment) for this. Or maybe a library function.

Anyway, this set of symbols looks somewhat arbitrary to me. According to [this](https://github.com/NixOS/nix/blob/a7540294cfae82c098e8691cd5212a9184add574/nix-rust/src/store/path.rs#L109) you might need to escape way more chars than this.

Reply:
Those are the characters that can appear in javascript package name, but can't appear in nix path name.

Reply:
Still would be nice to have some documentation.

I found [this](https://docs.npmjs.com/files/package.json) and it says that the package name can only contain “url-safe” characters. I’m not sure what this means, but I suppose, they refer to the “unreserved characters” from the URL RFC, namely:

```text
unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
```

which means that the only problematic one is `~` (which you did not include in your list). (Note that _both_ Nix and npm allow `.`, but not as the first char.) `@` and `/` in your list, I suppose, come from the scoped packages that have the form `@<scope>/<package>` and, according to the [doc](https://docs.npmjs.com/misc/scope) the restrictions for scope names are the same.

tldr; basically, it looks like you need to replace your list with just `~` and remove leading `@`.

Reply:
(Also, tbh, I would probably replace both `~` and `/` with `-` rather than `-` for readability reasons.)